### PR TITLE
Re-enable oc explain tests 

### PR DESCRIPTION
**Pull request by soltysh**
This picks #21739 and fixes verify problem.

/assign @juanvallejo @sttts

### CODE REVIEW

**Comment by juanvallejo** [comment]
e2e failure appears to be a flake

/test e2e-aws

File: vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go [hunk 1]
```diff
@@ -371,6 +371,11 @@ func (s *GenericAPIServer) InstallLegacyAPIGroup(apiPrefix string, apiGroupInfo
 	return nil
 }
 
+func (s *GenericAPIServer) RemoveOpenAPIData() {
```
Comment (sttts) [inline vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go]: this needs docs. Why do we need this? I remember we found out at KubeCon, but I already forgot.

File: pkg/cmd/openshift-apiserver/openshiftapiserver/openshift_apiserver.go [hunk 2]
```diff
@@ -509,6 +511,28 @@ func (c *completedConfig) withUserAPIServer(delegateAPIServer genericapiserver.D
 	return server.GenericAPIServer, &legacyStorageVersionMutator{version: userapiv1.SchemeGroupVersion, storage: storage}, nil
 }
 
+func (c *completedConfig) withOpenAPIAggregator(delegatedAPIServer *genericapiserver.GenericAPIServer) (*genericapiserver.GenericAPIServer, error) {
+	delegatedAPIServer.RemoveOpenAPIData()
```
Comment (sttts) [inline pkg/cmd/openshift-apiserver/openshiftapiserver/openshift_apiserver.go]: why is this called? This needs extensive docs. We looked into the why at KubeCon.

File: pkg/cmd/openshift-apiserver/openshiftapiserver/openshift_apiserver.go [hunk 3]
```diff
@@ -545,6 +569,12 @@ func (c completedConfig) New(delegationTarget genericapiserver.DelegationTarget,
 		return nil, err
 	}
 
+	// add aggregator last
```
Comment (sttts) [inline pkg/cmd/openshift-apiserver/openshiftapiserver/openshift_apiserver.go]: why last? We don't add an apiserver anymore. This just adds a nother controller. Maybe rename it to  `withOpenAPIAggregationController`

File: vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go (hunk 1 above)
Comment (juanvallejo) [reply vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go]: If we do not nil out existing openapi fields for the head of the delegation chain, adding the openapi aggregation controller won't work properly / at all

File: pkg/cmd/openshift-apiserver/openshiftapiserver/openshift_apiserver.go (hunk 2 above)
Comment (juanvallejo) [reply pkg/cmd/openshift-apiserver/openshiftapiserver/openshift_apiserver.go]: commented above, but the apiserver passed to the openapi aggregation controller must not have openapi fields set

**Pull request closed without merging**